app/src/main/java/fr/free/nrw/commons/LocationPicker/LocationPickerActivity.java

```diff
@@ -365,12 +370,21 @@ private void enableLocationComponent(@NonNull final Style loadedMapStyle) {
             // Set the component's render mode
             locationComponent.setRenderMode(RenderMode.NORMAL);
 
+            /** User's last location is unknown in case the access to location is turned off
+             * Ask for location access in case it is unavailable
+             */
+            if(locationComponent.getLastKnownLocation() == null){
+                showLocationOffDialog();
```

Comment:
IIUC, this particular invocation is getting invoked at all as the `if` statement above ...

```java
        if (PermissionsManager.areLocationPermissionsGranted(this))
```

... would be evaluated to false when location is not enabled. You should look into handling this differently. Further, I'm not sure `enableLocationComponent` is the correct method to place this check since it is invoked during the initialization of the activity. This results in the location request dialog being show even before the user hits the "target" icon. It would be better if we only request for the permission when the user actually hits the "target" icon.

Reply:
It is getting invoked even when the if statement above checks it beforehand. I had added it for the case where we don't have the last known location. I'd used `enableLocationComponent` since most of the code was redundant. It indeed shows up before tapping the target icon and shows the message accordingly. I'll make the changes you've suggested and request for the permission only on tapping the target icon

Reply:
Are you getting prompted with the location request dialog when the app does not have the location permission. I tested with this MR and I'm not shown the dialog when hitting the target icon.

To me, this seems to be due to what I've mentioned just above.

Reply:
Yes, I am getting the dialog again in case I deny in the beginning and use the target button later:

https://user-images.githubusercontent.com/83745993/232208929-483e57f4-4f7d-4a83-ad81-cce50a312d90.mp4

It doesn't ask for it again in case I allow in the beginning itself.

Reply:
@sivaraam, I have defined a separate function for this purpose and removed the dialog in the beginning. Now it asks for permission only on tapping the target icon.